lib/pure/math.nim

```diff
@@ -1211,6 +1211,41 @@ func `^`*[T: SomeNumber](x: T, y: Natural): T =
         break
       x *= x
 
+func isInteger(y: SomeFloat): bool =
+  ## Determines if a float represents an integer
+  ## Note this might trigger a change of the rounding mode.
+  # In C++, we usually use the `rint` function.
+  return round(y) == y
+
+func `^`*[T: SomeNumber, U: SomeFloat](x: T, y: U): float =
+  ## Computes `x` to the power of `y`.
+  ##
+  ## Error handling follows C++ specification
+  ## https://en.cppreference.com/w/cpp/numeric/math/pow
+  let
+    isZero_x: bool = (x == 0.0 or x == -0.0)
+    isNegZero: bool = classify(x) == fcNegZero
+    isPosZero: bool = classify(x) == fcZero
+    yIsFinite: bool = (y != Inf and y != -Inf)
+    yIsOddInteger: bool = (isInteger(y) and yIsFinite and (abs(int(y) mod 2) == 1))
+
+  assert not(isPosZero and y < 0 and yIsOddInteger)
+  assert not(isNegZero and y < 0 and yIsOddInteger)
+  assert not(isZero_x and y < 0 and y != -Inf)
+  assert not(isZero_x and y == -Inf)
+  assert not(x < 0 and not isInteger(x) and yIsFinite and not yIsOddInteger)
+  when defined(js):
+    # JS behavior follows an old version of IEEE 754 for compatibility reasons
+    # See https://262.ecma-international.org/#sec-numeric-types-number-exponentiate
```

Comment:
I might put this in the docs using two `#` signs. Do you think it should stay a simple comment here?